future/waitForConfirmation.go

```diff
@@ -0,0 +1,62 @@
+package future
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand-sdk/client/v2/common"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+)
+
+// WaitForConfirmation waits for a pending transaction to be accepted by the network
+type WaitForConfirmation struct {
+	c *algod.Client
+
+	// The ID of the pending transaction to wait for
+	txid string
+
+	// The number of rounds to block before existing with an error. If zero, there is no timeout
+	waitRounds uint64
+}
+
+// Do performs the HTTP request
+func (s *WaitForConfirmation) Do(ctx context.Context, headers ...*common.Header) (response models.NodeStatus, err error) {
```

Comment:
```suggestion
func WaitForConfirmation(c *algod.Client, txid string, waitRounds uint64, ctx context.Context, headers ...*common.Header) (response models.PendingTransactionInfoResponse, err error) {
```

Can you change the signature to this? Since this is no longer part of the algod package, it can just be a standalone function.